pyaqueduct/experiment.py

```diff
@@ -79,14 +82,22 @@ def tags(self) -> List[str]:
         return self._client.get_experiment(self.id).tags
 
     @validate_call
-    def add_tag(self, tag: str = Field(..., max_length=_MAX_TAG_LENGTH)) -> None:
-        """Add new tag to experiment."""
-        if not is_valid_tag(tag):
-            raise ValueError(
-                f"Tag {tag} should only contain alphanumeric characters, underscores or hyphens"
-            )
-
-        self._client.add_tag_to_experiment(experiment_uuid=self.id, tag=tag)
+    def add_tags(self, tags: List[TagString]) -> None:
+        """Add new tags to experiment.
+
+        Args:
+            tags: List of tags to be added to the experiment.
+
+        """
+
+        for tag in tags:
+            if not is_valid_tag(tag):
+                raise ValueError(
+                    f"Tag {tags} should only contain alphanumeric characters, "
```

Comment:
```suggestion
                    f"Tag {tag} should only contain alphanumeric characters, "
```

Reply:
I wanted to show all of the tags, but now looking at it again, it is better to return the problematic tag. Thanks.